crypto/bn/asm/rsaz-3k-avx512.pl

```diff
@@ -48,8 +48,17 @@
     $avx512ifma = ($1==2.11 && $2>=8) + ($1>=2.12);
 }
 
-if (!$avx512 && `$ENV{CC} -v 2>&1` =~ /((?:clang|LLVM) version|.*based on LLVM) ([0-9]+\.[0-9]+)/) {
-    $avx512ifma = ($2>=7.0);
+if (!$avx512 && `$ENV{CC} -v 2>&1`
+    =~ /(Apple)?\s*((?:clang|LLVM) version|.*based on LLVM) ([0-9]+)\.([0-9]+)\.([0-9]+)?/) {
```

Comment:
(now that this magic invocation is spreading, it seems like a "compiler-detect" function might be a useful abstraction somewhere)

Reply:
> (now that this magic regex is spreading, it seems like a "compiler-detect" function might be a useful abstraction somewhere)

Yeah, but that would be for a different PR